test/hotspot/gtest/runtime/test_signals.cpp

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (c) 2022, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+#include "precompiled.hpp"
+#ifndef _WIN32
+#include "runtime/os.hpp"
+#include "utilities/ostream.hpp"
+#include "unittest.hpp"
+
+#include <errno.h>
+#include <signal.h>
+#include <stdio.h>
+#include <sys/ucontext.h>
+#include <string.h>
+
+extern "C" {
+  static void sig_handler(int sig, siginfo_t *info, ucontext_t *context) {
+    printf( " HANDLER (1) " );
+  }
+}
+
+class PosixSignalTest : public ::testing::Test {
+  public:
+
+  static void check_handlers() {
+    struct sigaction act, old_SIGFPE_act, old_SIGILL_act;
+    act.sa_handler = (void (*)(int))sig_handler;
+    sigemptyset(&act.sa_mask);
+    act.sa_flags = 0;
+    ASSERT_NE(sigaction(SIGFPE, &act, &old_SIGFPE_act), -1) << "Setting SIGFPE handler failed (errno)";
+    ASSERT_NE(sigaction(SIGILL, &act, &old_SIGILL_act), -1) << "Setting SIGILL handler failed (errno)";
```

Comment:
Does that actually work? I would have thought that just prints the literal "errno". I meant something like this:

https://github.com/openjdk/jdk/blob/081c89316a2f35cad0a5dcc93e88707530e03d72/test/hotspot/gtest/logging/logTestUtils.inline.hpp#L56

Reply:
That was rather stupid of me.  Commit 4 contains your suggested errno change.

Reply:
No problem. For a moment I wondered if that was some formatting magic I didn't know :)